### DISCUSSION TITLE
Ahlborn-Galanti SysEx for Pedalboard as MIDI Controller

### DISCUSSION THREAD

**Discussion by jonredrico** [GrandOrgue/grandorgue#1877] [Help]
Hi everyone, I'm trying to connect my Galanti pedalboard to my MacBook as a MIDI controller. I was informed by Galanti that the pedalboard does not output MIDI through its DB15 connector. Instead it sends Galanti SysEx to the console which is then processed into MIDI. However, I don't have the console - is there a way I can utilize the DB-15 port to connect the pedalboard to my Mac as a MIDI Controller?

A search led me to the GrandOrgue changelog:

> 0.3.1.2323 (2019-05-04)
> Ahlborn-Galanti-Sysex

What does this entry mean? Would there be a chance that GrandOrgue is capable of interpreting Galanti Sysex if I'm able to figure out how to make a connection from the DB-15 port on my pedalboard to the USB-C port on my MacBook? Thanks everyone!

- Jonathan

**Comment by Aduragbemi-Adeyinka** [comment]
Last year I converted a pedalboard from a Gem Plenum (similar to Alhorn Galanti, or same but with a different name for another country). I did not use it's own electronics because the switches were totally bad, and I reworked the parallel pedal to a radiating one, so the pedal ends didn't  fit the electronics anymore. But I noted that they are matrix wired, that is, group of 8 notes to a common bus. You will notice that only 12 of the 15 pins are connected; 4 busses and 8 inputs. 

If your pedal switches are still ok and you're not changing anything, all you will need is an encoder that will generate the MIDI. you can use an Arduino Uno loaded with this code from my teacher; https://sites.google.com/site/casavantopus400/arduinouno
Then you can connect with  a MIDI to USB converter to your computer. 

Cheers.

**Comment by jonredrico** [comment]
Hi there, thank you for sharing your experience and code! It gives me hope that you were able to MIDIfy the pedalboard. I’ll do more reading and see what I can come up with. Thank you for your help!
